src/main/java/spoon/support/compiler/jdt/JDTBasedSpoonCompiler.java

```diff
@@ -542,43 +534,45 @@ protected void generateProcessedSourceFilesUsingCUs() {
 			CtPackage pack = element.getPackage();
 
 			// create package directory
-			File packageDir;
-			if (pack.isUnnamedPackage()) {
-				packageDir = new File(outputDirectory.getAbsolutePath());
-			} else {
-				// Create current package directory
-				packageDir = new File(outputDirectory.getAbsolutePath() + File.separatorChar + pack.getQualifiedName().replace('.', File.separatorChar));
-			}
-			if (!packageDir.exists()) {
-				if (!packageDir.mkdirs()) {
-					throw new RuntimeException("Error creating output directory");
-				}
-			}
+			Path packageDir = getPackageDir(outputDirectory, pack);
+			createDirectories(packageDir);
 
 			// print type
-			try {
-				File file = new File(packageDir.getAbsolutePath() + File.separatorChar + element.getSimpleName() + DefaultJavaPrettyPrinter.JAVA_FILE_EXTENSION);
-				file.createNewFile();
-
-				// the path must be given relatively to to the working directory
-				try (InputStream is = getCompilationUnitInputStream(cu);
-					FileOutputStream outFile = new FileOutputStream(file)) {
-					is.transferTo(outFile);
-				}
-
-				if (!printedFiles.contains(file)) {
-					printedFiles.add(file);
-				}
-
+			String fileName = element.getSimpleName() + DefaultJavaPrettyPrinter.JAVA_FILE_EXTENSION;
+			Path file = packageDir.resolve(fileName);
+			// the path must be given relatively to the working directory
```

Comment:
```suggestion
```
We can omit this now, I don't believe this was ever true. The code always passed an absolute path anyways.

Reply:
I did some code archaeology and this was introduced by martin in https://github.com/INRIA/spoon/commit/3a88497f7aeed29d2c195d52600d5b81c328515c. It became irrelevant in https://github.com/INRIA/spoon/commit/825c54619af4df02de3a8dd3fde2e8843a1be044.

I assume the compilation unit map used relative path names as key and that's why the comment was there.